Iris-Pipeline/steps/train.py

```diff
@@ -0,0 +1,197 @@
+    Afterwards it also logs the model using sklearn's own function.
+        This is done through MLFlow.sklearn.
+
+    Args:
+        dtc_model (DecisionTreeClassifier): Trained model from train_model()
+    """
+    # Creates the tree topology
+    fig, axes = plt.subplots(nrows=1, ncols=1, figsize=(4, 4), dpi=300)
+    tree.plot_tree(dtc_model)
+
+    fig.savefig(tree_plot_path)
+    
+    return tree_plot_path
+
+
+@click.command()
+@click.option("--process_run_id")
+@click.option("--graphs")
+@click.option("--config_path")
+def task(process_run_id, graphs, config_path) -> None:
+    """
+    Task function that orchestrates the training step
+
+    Task function organises the training step.
+    Following functions and their order of execution:
+        Parameter_tuning(): Checks for the best possible parameters,
+            to use on the DecisionTreeClassifier
+        Model_creation(): Creates the model using the variables,
+            from Parameter_tuning()
+        K_fold_cross_validation(): Cross validates on the data and model
+        Train_model(): Trains the model using the split dataframes,
+            that was further split into 4 Pandas series
+        Save_model(): saves the trained model using the Sklearn save_model function
+
+    Args:
+        process_run_id (str): the run id from the process step passed through main step
+    """
+    with mlflow.start_run() as mlrun:
+        with open(config_path, "r", encoding="UTF-8") as ymlfile:
+            cfg = yaml.load(ymlfile, Loader=yaml.FullLoader)
+
+        process_run = mlflow.tracking.MlflowClient().get_run(process_run_id)
+
+        version_number = mlrun.info.run_id[:5]
+
+        df_path = Path(process_run.info.artifact_uri, cfg["train_data_name"])
+        df = pd.read_csv(df_path)
+
+        X = df[cfg["features"].keys()]
+        y = df[cfg["label_name"]]
+
+        max_depth, criterion, splitter = parameter_tuning(
+            nr_kfold=cfg["kfold_settings"]["nr_splits"],
+            X_train=X,
+            y_train=y,
+        )
+
+        dtc_model = DecisionTreeClassifier(
+            criterion=criterion,
+            splitter=splitter,
+            max_depth=max_depth,
+        )
+
+        indices_kfold = StratifiedKFold(
+            n_splits=cfg["kfold_settings"]["nr_splits"],
+            shuffle=cfg["kfold_settings"]["shuffle"],
+        )  
+         
+        for train_index, test_index in indices_kfold.split(X, y):
+            train_model(
+                dtc_model=dtc_model,
+                X=X,
+                y=y,
+                train_index=train_index,
+                test_index=test_index
+            )
+
+        tree_plot_path = plot_tree(
+            dtc_model=dtc_model,
+            tree_plot_path=cfg["tree_plot_path"],
+        )
+        
+        mlflow.log_artifact(tree_plot_path)
+
+        mlflow.sklearn.log_model(
+            sk_model=dtc_model,
+            artifact_path=cfg["model_name"]+f"-{version_number}",
```

Comment:
Correct, it was only because it made it easy to fetch the model from the model registry. I’ll have to rethink how to name them so i can access them still. I can f.x only have one model and just read whatever is in the models folder (since it would only have one model)